### PR TITLE
feat: Add detail to README + add createPlanets to local game

### PR DESCRIPTION
**Pull request by cha0sg0d**
chore: Add game:createPlanets to local game deployment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,74 @@
...
-## Install
-
-- Clone with submodules `git clone --recurse-submodules https://github.com/projectsophon/darkforest-local` or if you've already cloned and want to update the module `git submodule update --init --recursive`
-- Install [Yarn](https://classic.yarnpkg.com/en/docs/install)
-- Run `yarn` at the top level to automatically install all dependencies from subfolders.
-
+The Dark Forest client deploy process is set up for players to easily fork the client and connect to
+the mainnet game, but that design decision currently makes it difficult to run a playable version
+of the game locally. This repository provides a setup for running a local game with just a few
+steps.
```
Comment (phated) [inline README.md:6]:
```suggestion
The Dark Forest client deploy process is set up for players to easily fork the client and connect to the mainnet game, but that design decision currently makes it difficult to run a playable version of the game locally. This repository provides a setup for running a local game with just a few steps.
```

File: README.md [hunk 2]
```diff
@@ -1,25 +1,74 @@
...
+
+## Builder's Guide
+
+For a full tutorial on using `darkforest-local` with multiple **quests**, check out the [Community Round Builder's Guide](builders_guide.md).
```
Comment (phated) [inline README.md:10]:
```suggestion
```

We'll add this section in the Builders Guide PR.

File: README.md [hunk 3]
```diff
@@ -1,25 +1,74 @@
...
+
+## How darkforest-local Works
```
Comment (phated) [inline README.md:12]:
```suggestion
## How `darkforest-local` works
```

File: README.md [hunk 4]
```diff
@@ -1,25 +1,74 @@
...
+
+This repo uses [submodules](.gitmodules) to pair the [Dark Forest Ethereum
+backend](https://github.com/darkforest-eth/eth) with the [Dark Forest TypeScript
+frontend](https://github.com/darkforest-eth/client) so you can launch a local game.
```
Comment (phated) [inline README.md:16]:
```suggestion
This repo uses [submodules](.gitmodules) to pair the [Dark Forest Ethereum backend](https://github.com/darkforest-eth/eth) with the [Dark Forest TypeScript frontend](https://github.com/darkforest-eth/client) so you can launch a local game.
```

File: README.md [hunk 5]
```diff
@@ -1,25 +1,74 @@
...
+
+It also uses Yarn, a package manager and that allows for multiple [workspaces](https://classic.yarnpkg.com/lang/en/docs/workspaces) to exist within a project.
```
Comment (phated) [inline README.md:18]:
```suggestion
It also uses Yarn, a package manager (like `npm`) that allows for multiple [workspaces](https://classic.yarnpkg.com/lang/en/docs/workspaces) to exist within a project.
```

File: README.md [hunk 6]
```diff
@@ -1,25 +1,74 @@
...
+
+The workspaces allows each [submodule](.gitmodules) to have their own packages and configuration. 
+Yarn places all of the packages for each submodule in the top level `node_modules/` folder.
+
+## Requirements
+* Install [Yarn](https://classic.yarnpkg.com/en/docs/install)
+* Install `node >= 14` [nvm](https://github.com/nvm-sh/nvm)
```
Comment (phated) [inline README.md:25]:
```suggestion
* Install `node >= 14` (You can consider using [nvm](https://github.com/nvm-sh/nvm))
```

File: README.md [hunk 7]
```diff
@@ -1,25 +1,74 @@
...
+
+### Fastest Method for Running a Local Game
```
Comment (phated) [inline README.md:27]:
```suggestion
### Quickstart for running a local game
```

File: README.md [hunk 8]
```diff
@@ -1,25 +1,74 @@
...
+1. Fork [darkforest-local](https://github.com/projectsophon/darkforest-local) to your Github repo.
```
Comment (phated) [inline README.md:28]:
```suggestion
1. Fork [darkforest-local](https://github.com/projectsophon/darkforest-local) to your GitHub account
```

File: README.md [hunk 9]
```diff
@@ -1,25 +1,74 @@
...
+2. `git clone https://github.com/<your_name>/darkforest-local.git`
```
Comment (phated) [inline README.md:29]:
```suggestion
2. `git clone --recurse-submodules https://github.com/<your_name>/darkforest-local.git`
```

File: README.md [hunk 10]
```diff
@@ -1,25 +1,74 @@
...
+3. `git submodule update --init --recursive --remote --merge`
```
Comment (phated) [inline README.md:30]:
```suggestion
3. If you didn't clone with `--recurse-submodules` or already have a cloned version: `git submodule update --init --recursive --remote --merge`
```

File: README.md [hunk 11]
```diff
@@ -1,25 +1,74 @@
...
+4. `yarn`
+5. `yarn start`
+
+### Better Method for Running a Local Game
```
Comment (phated) [inline README.md:34]:
```suggestion
### If you plan to make changes to `darkforest-local`
```

File: README.md [hunk 12]
```diff
@@ -1,25 +1,74 @@
...
+1. Fork [darkforest-local](https://github.com/projectsophon/darkforest-local) to your Github repo.
+2. Fork [darkforest-eth/eth](https://github.com/darkforest-eth/eth) to your Github repo
+3. Fork [darkforest-eth/client](https://github.com/darkforest-eth/client) to your Github repo.
```
Comment (phated) [inline README.md:37]:
```suggestion
1. Fork [darkforest-local](https://github.com/projectsophon/darkforest-local) to your GitHub account
2. Fork [darkforest-eth/eth](https://github.com/darkforest-eth/eth) to your GitHub account
3. Fork [darkforest-eth/client](https://github.com/darkforest-eth/client) to your GitHub account
4. Fork [darkforest-eth/circuits](https://github.com/darkforest-eth/circuits) to your GitHub account
5. Fork [darkforest-eth/packages](https://github.com/darkforest-eth/packages) to your GitHub account
```

File: README.md [hunk 13]
```diff
@@ -1,25 +1,74 @@
...
+4. Clone your darkforest-local repo: `git clone https://github.com/<your_name>/darkforest-local.git`
+5. Update the `.gitmodules` file to point to your new forks of `eth` and `client.
```
Comment (phated) [inline README.md:39]:
```suggestion
5. Update the `.gitmodules` file to point to your new forks of `eth`, `client`, `circuits`, and `packages` 
```

File: README.md [hunk 14]
```diff
@@ -1,25 +1,74 @@
...
+1. Fork [darkforest-local](https://github.com/projectsophon/darkforest-local) to your Github repo.
+2. Fork [darkforest-eth/eth](https://github.com/darkforest-eth/eth) to your Github repo
+3. Fork [darkforest-eth/client](https://github.com/darkforest-eth/client) to your Github repo.
+4. Clone your darkforest-local repo: `git clone https://github.com/<your_name>/darkforest-local.git`
+5. Update the `.gitmodules` file to point to your new forks of `eth` and `client.  
+    ex: 
+    * `url = https://github.com/darkforest-eth/eth` =>  
+    `url = https://github.com/cha0sg0d/eth`
+    * `url = https://github.com/darkforest-eth/client` =>  
+    `url = https://github.com/cha0sg0d/client`
```
Comment (phated) [inline README.md:44]:
```suggestion
    * `url = https://github.com/darkforest-eth/eth` => `url = https://github.com/cha0sg0d/eth`
    * `url = https://github.com/darkforest-eth/client` => `url = https://github.com/cha0sg0d/client`
    * `url = https://github.com/darkforest-eth/circuits` => `url = https://github.com/cha0sg0d/circuits`
    * `url = https://github.com/darkforest-eth/packages` => `url = https://github.com/cha0sg0d/packages`
```

File: README.md [hunk 15]
```diff
@@ -1,25 +1,74 @@
...
+    2. `cd client`
+        1. `git checkout -b <new_name>`
+7. Install packages and dependencies
+    * `yarn`
+8. Start a game
+    * `yarn start`
+
+        If `yarn start` returns the error `ERR_OSSL_EVP_UNSUPPORTED`:  
+        * *This error occurs when using Node v17+*
+        * To fix, run `export NODE_OPTIONS=--openssl-legacy-provider` or switch to `14 <= node <= 16`
```
Comment (phated) [inline README.md:60]: Do we need this? Odd version node aren't recommended for use, as they are canary builds that break all the time.

File: README.md [hunk 16]
```diff
@@ -40,9 +89,18 @@ Unlike in development mode, in production mode you will need to create a `.env`
 `client/` and `eth/` submodules. You can find the set of environment variables you will need to
 populate those `.env` files with in their adjacent `.env.example` files.
 
-> **Danger!** The `.env` files ARE in the respective `.gitignore`s of the aforementioned submodules,
+> Danger! The `.env` files ARE in the respective `.gitignore`s of the aforementioned submodules,
```
Comment (phated) [inline README.md:92]:
```suggestion
> **Danger!** The `.env` files ARE in the respective `.gitignore`s of the aforementioned submodules,
```

File: README.md [hunk 17]
```diff
@@ -53,28 +111,47 @@ You should see something like this:
...
-```
-
-You should see something like this:
-
-![yarn workspace eth hardhat:prod deploy](img/netlify_prod_deploy.png)
-
-### Step 3: allow player addresses
+To use Netlify:
+
+- Make a new [Netlify account](https://app.netlify.com/signup) using your Github profile.
```
Comment (phated) [inline README.md:125]:
```suggestion
- Make a new [Netlify account](https://app.netlify.com/signup) using your GitHub account
```

File: README.md [hunk 18]
```diff
@@ -53,28 +111,47 @@ You should see something like this:
...
+- [Import](https://app.netlify.com/start) your Github client repo as a new Netlify site.
```
Comment (phated) [inline README.md:126]:
```suggestion
- [Import](https://app.netlify.com/start) your `client` repository as a new Netlify site.
```

File: README.md [hunk 19]
```diff
@@ -53,28 +111,47 @@ You should see something like this:
...
+- Login to your account
+    - `netlify login`
+- Connect to your app
+    - `netlify link`
+    
+    ![link](img/link.png)
+    
+- Choose *Enter a site ID*
+    
+    ![Screen Shot 2022-01-15 at 12.35.15 PM.png](img/site_id.png)
```
Comment (phated) [inline README.md:138]:
```suggestion
    ![Site ID](img/site_id.png)
```

File: README.md [hunk 20]
```diff
@@ -53,28 +111,47 @@ You should see something like this:
...
+- Make a new [Netlify account](https://app.netlify.com/signup) using your Github profile.
+- [Import](https://app.netlify.com/start) your Github client repo as a new Netlify site.
+- Install the Netlify CLI
+    - `npm install netlify-cli -g`
+- Login to your account
+    - `netlify login`
+- Connect to your app
+    - `netlify link`
+    
+    ![link](img/link.png)
```
Comment (phated) [inline README.md:134]:
```suggestion
    ![Netlify link](img/link.png)
```

File: README.md [hunk 21]
```diff
@@ -53,28 +111,47 @@ You should see something like this:
...
+- Input the id that you find by clicking on your new site on Netlify. In this example, the id is `jolly-hermann-3947ca`
+- Now that your site is linked, you can deploy to Netlify from the CLI.
+    - `yarn workspace client deploy:prod`
+    - If successful, you will see something like this.
+        
+        ![deploy_success](img/netlify_deploy.png)
+        
+    - *Note: To do this, you will have to turn on automatic deploys for your site. This means that if you push an update to your main Git branch, Netlify will attempt to redeploy the site. This will fail, because the client relies on the most recent versions of the packages you have created, but Netlify will try to install the older versions from npm instead of locally.*
+        
+        ![Screen Shot 2022-01-15 at 12.38.57 PM.png](img/prod_deploy.png)
```
Comment (phated) [inline README.md:149]:
```suggestion
        ![deploy_failures](img/prod_deploy.png)
```

File: README.md [hunk 22]
```diff
@@ -53,28 +111,47 @@ You should see something like this:
...
+    ![Screen Shot 2022-01-15 at 12.35.15 PM.png](img/site_id.png)
+    
+- Input the id that you find by clicking on your new site on Netlify. In this example, the id is `jolly-hermann-3947ca`
+- Now that your site is linked, you can deploy to Netlify from the CLI.
+    - `yarn workspace client deploy:prod`
+    - If successful, you will see something like this.
+        
+        ![deploy_success](img/netlify_deploy.png)
+        
+    - *Note: To do this, you will have to turn on automatic deploys for your site. This means that if you push an update to your main Git branch, Netlify will attempt to redeploy the site. This will fail, because the client relies on the most recent versions of the packages you have created, but Netlify will try to install the older versions from npm instead of locally.*
```
Comment (phated) [inline README.md:147]: Why don't you turn off the deploys? Or instead recommend the users to deploy a subdirectory of darkforest-local?

File: README.md [hunk 23]
```diff
@@ -112,13 +186,19 @@ Then, once you've found a home planet, you should be able to enter the game!
...
+### Step 4: Play your game
+
+- Now go the URL given by Netlify → something like [https://jolly-hermann-3947ca.netlify.app/](https://jolly-hermann-3947ca.netlify.app/) and test out your game! Login with one of the private keys for the addresses that were whitelisted. Make some moves and test that everything works as expected.
+
 ## Update to latest Dark Forest code
 
-If theres been new Dark Forest updates released and we haven't yet updated this repo yet, it is possible you can get away with updating yourself by running `git submodule update --remote --merge` and remember to run `yarn` again.
+If theres been new Dark Forest updates released and we haven't yet updated this repo yet, it is
+possible you can get away with updating yourself by running `git submodule update --remote --merge`
+and remember to run `yarn` again.
```
Comment (phated) [inline README.md:197]:
```suggestion
If theres been new Dark Forest updates released and we haven't yet updated this repo yet, it is possible you can get away with updating yourself by running `git submodule update --remote --merge` and remember to run `yarn` again.
```

**Review by phated — CHANGES_REQUESTED** [review]
This is looking great. Just wanted to do some cleanup. The one big question I had was about netlify deploys. I think if you use the darkforest-local repository, you can deploy a subdirectory and then you wouldn't have an issue with deployed packages.

File: README.md (hunk 11 above)
Comment (cha0sg0d) [reply README.md:26]: I link to Better Method for Running a Local Game in the builder's guide so will need to update this

File: README.md (hunk 15 above)
Comment (cha0sg0d) [reply README.md:52]: good to know, let's remove

File: README.md [hunk 24]
```diff
@@ -2,24 +2,67 @@
...
+This repo uses [submodules](.gitmodules) to pair the [Dark Forest Ethereum backend](https://github.com/darkforest-eth/eth) with the [Dark Forest TypeScript frontend](https://github.com/darkforest-eth/client) so you can launch a local game.
+
+It also uses Yarn, a package manager (like `npm`) that allows for multiple [workspaces](https://classic.yarnpkg.com/lang/en/docs/workspaces) to exist within a project.
+
+The workspaces allows each [submodule](.gitmodules) to have their own packages and configuration. 
+Yarn places all of the packages for each submodule in the top level `node_modules/` folder.
+
+## Requirements
+* Install [Yarn](https://classic.yarnpkg.com/en/docs/install)
+* Install `node >= 14` (You can consider using [nvm](https://github.com/nvm-sh/nvm))
```
Comment (phated) [inline README.md:17]:
```suggestion

* Install `node >= 14` (You can consider using [nvm](https://github.com/nvm-sh/nvm))
* Install [Yarn](https://classic.yarnpkg.com/en/docs/install)
```

Let's reorder these.

File: README.md [hunk 25]
```diff
@@ -2,24 +2,67 @@
...
+    * `url = https://github.com/darkforest-eth/circuits` => `url = https://github.com/cha0sg0d/circuits`
+    * `url = https://github.com/darkforest-eth/packages` => `url = https://github.com/cha0sg0d/packages`
+6. Fetch the code from the submodules
+    * `git submodule update --init --recursive`   
+7. Add new branches for developing:
+    - The `darkforest-local` monorepo detaches the submodules from their current HEADs. If you want to save your changes (for example, if you're testing an new contract in `eth`), you'll need to make a new branch in these submodules.
+    1. `cd eth`
+        1. `git checkout -b <new_name>`
+    2. `cd client`
+        1. `git checkout -b <new_name>`
```
Comment (phated) [inline README.md:46]:
```suggestion
    2. `cd client`
        1. `git checkout -b <new_name>`
    3. `cd circuits`
        1. `git checkout -b <new_name>`
    4. `cd packages`
        1. `git checkout -b <new_name>`
```
Probably should include the other submodules.

File: README.md [hunk 26]
```diff
@@ -2,24 +2,67 @@
...
+7. Install packages and dependencies
+    * `yarn`
+8. Start a game
+    * `yarn start`
+
+        If `yarn start` returns the error `ERR_OSSL_EVP_UNSUPPORTED`:  
+        * *This error occurs when using Node v17+*
+        * To fix, run `export NODE_OPTIONS=--openssl-legacy-provider` or switch to `14 <= node <= 16`
```
Comment (phated) [inline README.md:54]:
```suggestion
```

File: netlify.toml [hunk 27]
```diff
@@ -0,0 +1,14 @@
...
+
+[build.environment]
+  CONVERSATION_API_HOST = ""
+  LEADERBOARD_API = ""
+  WEBSERVER_URL = ""
+
+[[redirects]]
+    from = "/*"
+    to = "/index.html"
+    status = 200
```
Comment (phated) [inline netlify.toml:14]:
```suggestion
  from = "/*"
  to = "/index.html"
  status = 200
```
Can this have the same indentation as the above?

**Review by phated — APPROVED** [review]
Excellent! Thanks @cha0sg0d 🎉

**Pull request merged**